lib/common/xml.c

```diff
@@ -1489,15 +1487,16 @@ pcmk__xe_log(int log_level, const char *file, const char *function, int line,
     name = crm_element_name(data);
 
     if (pcmk_is_set(options, xml_log_option_open)) {
-        char *buffer = NULL;
+        GString *buffer = g_string_sized_new(128);
 
-        insert_prefix(options, &buffer, &offset, &max, depth);
+        insert_prefix(options, buffer, depth);
 
         if (data->type == XML_COMMENT_NODE) {
-            buffer_print(buffer, max, offset, "<!--%s-->", data->content);
+            g_string_append_printf(buffer, "<!--%s-->",
+                                   (const gchar *) data->content);
```

Comment:
It just dawned on me that this function should be called pcmk__xml_log() or pcmk__log_xml() since it handles comments as well as elements. It would be safer to explicitly ignore other node types.

Reply:
May be a tad confusing relative to `crm_log_xml.*()` but more accurate